lib/dom.js

```diff
@@ -4,6 +4,117 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */
+/* Crypto */
+
+// This is a helper type to simplify the specification, it isn't an interface
+// and there are no objects implementing it.
+// https://developer.mozilla.org/en-US/docs/Web/API/BufferSource
+type BufferSource = ArrayBuffer | $ArrayBufferView;
+
+type DOMString = string;
+
+declare interface Crypto extends RandomSource {
+  subtle: SubtleCrypto; // readonly
+}
+
+type CryptoKey$Type =
+  | 'secret'
+  | 'public'
+  | 'private';
+type CryptoKey$Usages =
+  | 'encrypt'
+  | 'decrypt'
+  | 'sign'
+  | 'verify'
+  | 'deriveKey'
+  | 'deriveBits'
+  | 'wrapKey'
+  | 'unwrapKey';
+declare interface CryptoKey {
+    algorithm: Object; // readonly
+    extractable: Boolean; // readonly
+    type: CryptoKey$Type; // readonly
+    usages: CryptoKey$Usages[]; // readonly
+}
+
+// TODO: CryptoKeyPair
+declare interface CryptoKeyPair {
+
+}
+
+// TODO: JSONWebKey
+type JSONWebKey = {};
+
+type RandomSource =  {
+  getRandomValues(typedArray: $TypedArray): void;
```

Comment:
The [MDN article](https://developer.mozilla.org/en-US/docs/Web/API/RandomSource/getRandomValues) uses mutating the array, but [w3.org](https://www.w3.org/TR/WebCryptoAPI/#Crypto-method-getRandomValues) officially specifies the array must also be returned.
MDN also depends on that feature in [the `encrypt` and `decrypt` examples](https://github.com/facebook/flow/pull/5027/commits/d56ffe786f3509999c304605e36e96979f04a3bd#r145700797).

TL;DR: I imagine it should go a little something like `getRandomValues<T: $TypedArray>(typedArray: T): T;`

Reply:
I just followed the [MDN spec](https://developer.mozilla.org/en-US/docs/Web/API/RandomSource/getRandomValues), but if [w3.org](https://www.w3.org/TR/WebCryptoAPI/#Crypto-method-getRandomValues) specifies it with a return-variable and MDN uses it in there examples, this one should defo have one too. 🙂